### ISSUE TITLE
[Grid - Basics] Buttons in Header and Menu Button do not have accessible names

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3183]
## Current behavior

In stories [Grid Basics Header](https://storybook.sanomalearning.design/?path=/story/grid-basics--header) and [Grid Basics Menu Button](https://storybook.sanomalearning.design/?path=/story/grid-basics--menu-button) buttons with three dots do not have accessible name provided with `aria-label` or `aria-labelledby`. They are announced by screen readers as "menu pop up collapsed, button" or "menu button collapsed subMenu" which is not sufficient.

<img width="833" height="148" alt="Image" src="https://github.com/user-attachments/assets/1d150e91-3a16-485f-8783-f85c64aa04b6" />

<img width="659" height="351" alt="Image" src="https://github.com/user-attachments/assets/85742208-bca2-4931-ac03-0c2038d1c4ce" />

## Suggested solution

It's recommended to add `aria-label` or `aria-labelledby` to each `button` that currently does not have accessible name.
Accessible name should be unique in the set of buttons, so if user only use Tab to go from button to button or lists all buttons that are inside grid they still will understand which button is responsible for which action. 
Proposed accessibility name could be: "Options for NAME SURNAME" or "Options NAME SURNAME" which is easier to achieve using `aria-labelledby`. 

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
